src/app/components/formElements/Toggle/ToggleInput.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { FocusEvent } from 'react';
+import styles from './toggle.module.css';
+import {
+  DefaultValue,
+  SelectOption,
+  ToggleInputProps,
+} from '@/app/interfaces/FormElements';
+import { FieldValues } from 'react-hook-form';
+import Label from '../Label/Label';
+import HelperText from '../HelperText/HelperText';
+import ErrorBox from '../ErrorBox/ErrorBox';
+import classNames from 'classnames';
+import useDefaultValue from '@/app/hooks/useDefaultValue';
+
+const ToggleInput = <TFieldValues extends FieldValues>({
+  name,
+  label,
+  showLabel = false,
+  options,
+  helperText,
+  required = false,
+  defaultValue = false as DefaultValue<TFieldValues>,
+  onChange,
+  orientation = 'horizontal',
+  defaultErrorMessage,
+  renderProps,
+  methods,
+}: ToggleInputProps<TFieldValues>) => {
+  const {
+    field,
+    formState: { errors },
+  } = renderProps;
+  const error = errors[name];
+  const { disabled, onBlur, value } = field;
+
+  useDefaultValue<TFieldValues>({
+    renderProps,
+    defaultValue,
+    setValue: methods.setValue,
+  });
+
+  const setValue = (value: SelectOption['value']) => {
+    field.onChange(value);
+    onChange?.(value);
+  };
+
+  return (
+    <div
+      className={styles.toggleRoot}
+      role='group'
+      aria-disabled={disabled}
+      onBlurCapture={(e: FocusEvent<HTMLDivElement, Element>) => {
+        !(e.currentTarget as Node)?.contains(e.relatedTarget as Node) &&
+          onBlur();
+      }}
+    >
+      {showLabel && (
+        <Label
+          name={name}
+          color={error && 'red'}
+          label={label}
+          required={required}
+        />
+      )}
+      <div className={classNames(styles.toggleContainer, styles[orientation])}>
+        {options.map(({ label, value: itemValue, label2 }) => (
+          <div
+            key={itemValue.toString()}
+            role='checkbox'
+            aria-disabled={disabled}
+            aria-checked={value === itemValue}
+            aria-label={label}
+            className={styles.switchContainer}
+          >
+            <label className={styles.switch}>
+              <input
+                type='checkbox'
+                disabled={disabled}
+                onChange={() => {
+                  setValue(value === itemValue);
+                }}
+              />
+              <span className={classNames(styles.slider, styles.round)}></span>
+            </label>
+            <label className={styles.label}>
+              {value === itemValue ? label : label2}
+            </label>
```

Comment:
This is a note for future reference. We will try to gradually enforce accessibilities. Following are the things that we need to keep in mind 
- Firstly, either we will use semantic HTML (<input type="checkbox" />) or provide meaningful `role` or `aria` attributes that provides semantics. (In this case, `role="checkbox"`)
- Secondly, the whole application needs to be accessible by keyboard. The focused/active element should have a distinguishing style so that it's clear where we are through our keyboard navigation.
- Lastly, need to attach proper `aria` attributes to components.

We will tackle these things later on, I will create cards for these. Giving you a brief idea for the time being.

However, don't use `<label />` if we the element is not meant to contain any type of labels.